src/main/java/frc/robot/constants/ScoringSetpoints.java

```diff
@@ -66,12 +69,88 @@ public static record ScoringSetpoint(String name, Distance elevatorHeight, Angle
   public final ScoringSetpoint ground =
       new ScoringSetpoint("ground", Meters.of(0.2), Rotations.of(-0.75));
 
+  /**
+   * Combined with variableL4WristAngles, these form the interpolate double used for variable
+   * distance L4 scoring
+   */
+  public final Distance[] variableL4AlongTrackDistances = {Meters.of(0.1), Meters.of(0.0)};
+
+  /**
+   * Combined with variableL4AlongTrackDistances, these form the interpolate double used for
+   * variable distance L4 scoring
+   */
+  public final Angle[] variableL4WristAngles = {Rotations.of(0.32), Rotations.of(0.09)};
+
+  /** Maps along track distances (in meters) to wrist angles (in rotations) */
+  public static InterpolateDouble variableL4WristMap;
+
+  /** Keep track of a reference to the drive subsystem to access along track distance */
+  @JSONExclude private Drive driveReference = null;
+
+  /**
+   * Give the ScoringSetpoints instance a reference to the drivetrain, used to obtain along track
+   * distance
+   *
+   * @param driveReference The Drive subsystem instance
+   */
+  public void setDriveReference(Drive driveReference) {
+    this.driveReference = driveReference;
+  }
+
+  /**
+   * Create the InterpolateDouble to map along track distances to wrist angle.
+   *
+   * <p>This must be called AFTER syncing the constants with JSON. This method fills the loaded
+   * constants on an INSTANCE of ScoringSetpoints. Make sure it is called on an instance that has
+   * synced its JSON constants already.
+   *
+   * @throws Exception If constants are incorrectly defined with different lengths, an exception
+   *     will be thrown
+   */
+  public void fillVariableL4WristMap() throws Exception {
+    HashMap<Double, Double> wristMap = new HashMap<>();
+
+    if (variableL4AlongTrackDistances.length != variableL4WristAngles.length) {
+      throw new Exception(
+          "Variable L4 Along Track distances and Wrist angles have different lengths. Must be a 1:1 mapping.");
+    }
+
+    for (int i = 0; i < variableL4AlongTrackDistances.length; i++) {
+      wristMap.put(
+          variableL4AlongTrackDistances[i].in(Meters), variableL4WristAngles[i].in(Rotations));
+    }
+
+    variableL4WristMap = new InterpolateDouble(wristMap);
+  }
+
+  /**
+   * Creates the L4 wrist map, except it is just the default L4 setpoint.
+   *
+   * <p>This can be used to fill the map if the constants are defined incorrectly, and must be
+   * called AFTER drivetrain constants are synced using JSON (it depends on a synced JSON value from
+   * drivetrainConstants)
+   */
+  public void fillPlaceholderL4WristMap() {
+    HashMap<Double, Double> wristMap = new HashMap<>();
+
+    wristMap.put(0.0, L4.wristAngle().in(Rotations));
+
+    variableL4WristMap = new InterpolateDouble(wristMap);
+  }
```

Comment:
See other comment about not failing into a placeholder, but rather the original single setpoint behavior